src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs

```diff
@@ -12,6 +12,32 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics
     [CompilerTrait(CompilerFeature.RefLocalsReturns)]
     public class RefLocalsAndReturnsTests : CompilingTestBase
     {
+        [Fact]
+        public void RefReassignIdentityConversion()
+        {
+            var comp = CreateStandardCompilation(@"
+class C
+{
+    void M()
+    {
+        string s = ""s"";
+        object o = s;
+        ref string rs = ref s;
+        ref object ro = ref o;
+
+        rs = ref (string)o;
+        ro = ref s;
```

Comment:
Add an ordinary assignment case - " ro = s" . That should not generate an error.